### DISCUSSION TITLE
有没有办法自定义不同版本node的全局和缓存路径，并且切换不同版本自动适应？

### DISCUSSION THREAD

**Discussion by KervinWen** [1111mp/nvm-desktop#79] [Q&A]
我在下载18LTS版本后，通过set prefix、cache的命令，指向了18版本中的node_global、node_cache文件夹。
下载20LTS版本并切换，查询prefix、cache路径没有发生变化
我知道这是设置了用户文件夹下的.npmrc文件。只是不知道能否实现或者有没有必要实现如题的功能

**Comment by 1111mp** [comment]
因为`nvm-desktop`在最开始设计的时候就是想尽可能地还原用户在不借助任何工具情况下，自己手动从官网下载不同`node`版本的安装文件安装到本地计算机的这一过程，所以这部分就直接和该过程的默认行为保持一致了。

我刚刚也思考过如何实现这个功能，目前的方案就是在执行`npm config set prefix/cache "path"`的时候记录下来，然后切换`node`版本的时候自动根据之前保存的信息调用`npm config set prefix/cache "path"`更新。但是对应用户来说，同个`node`版本下，可以使用`npm set config prefix/cache`多次设置不同的目录，然后根据情况自己去管理（不排除有用户需要这样），所以就不太好维护（虽然可以只记录最后一次的）。但是最后感觉这部分还是交给用户自己管理吧（和npm的默认行为一样），因为如果做得太多反而会影响用户的自由度。

所以对于你这种情况，你可以在切换版本之后自己再手动`npm config set prefix/cache`设置一下。

当然，这个功能可以先记录在这里，如果后面需要的人多了，可以支持这个功能的。还希望能够理解。

**Reply by KervinWen** [reply]
好的明白
主要之前使用nvm的时候，有symlink软连接用来指向当前版本nodejs
这种情况下，似乎prefix和cache统一设置为NVM_SYMLINK指向的文件夹下的node_global和node_cache就可以

**Reply by 1111mp** [reply]
其实如果你不设置npm的prefix/cache，那么它们的目录默认是在对应node版本下的（`%HOMEPATH%\.nvmd\versions\{version}`） 当你切换node版本时就不需要手动设置了

![image](https://github.com/1111mp/nvm-desktop/assets/31227919/214993f1-034a-4459-aab7-dc5cde723af3)

**Reply by KervinWen** [reply]
prefix这个我知道
只是我还想设置cache，才导致了一开始的问题
谢谢解答
